certification/internal/openshift/openshift_version.go

```diff
@@ -3,25 +3,31 @@ package openshift
 import (
 	"context"
 	"fmt"
-	"os"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
 
 	configv1Client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
 	log "github.com/sirupsen/logrus"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	ctrl "sigs.k8s.io/controller-runtime"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
-func GetOpenshiftClusterVersion() (runtime.OpenshiftClusterVersion, error) {
-	if _, ok := os.LookupEnv("KUBECONFIG"); !ok {
-		return runtime.UnknownOpenshiftClusterVersion(), fmt.Errorf("KUBECONFIG not specified")
+func GetOpenshiftClusterVersion(kubeconfig []byte) (runtime.OpenshiftClusterVersion, error) {
+	if len(kubeconfig) == 0 {
+		return runtime.UnknownOpenshiftClusterVersion(), fmt.Errorf("kubeconfig was not provided")
 	}
-	kubeConfig, err := ctrl.GetConfig() // TODO(): Identify how to get the config/rest client using an io.Reader or similar instead of this.
+
+	cc, err := clientcmd.NewClientConfigFromBytes(kubeconfig)
```

Comment:
Can you use this instead, and not have to do the next block? 
https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L120

It looks like it literally does both of these.

Reply:
Sure thing. The clientcmd package has way too many ways to do what feels like "the same thing". Totally missed this convenience function.

Reply:
Fixed